src/routes/error-router.ts

```diff
@@ -1,92 +1,94 @@
-		[Errors.IP_ALLOWLIST_MISCONFIGURED]: `The GitHub org you are trying to connect is currently blocking our requests. To configure the GitHub IP Allow List correctly, <a href="${envVars.GITHUB_REPO_URL}/blob/main/docs/ip-allowlist.md">please follow these instructions</a>.`,
-		[Errors.MISSING_GITHUB_APP_NAME]: "There was a problem creating your GitHub App. Please make sure you filled the GitHub App name and try again."
-	};
-
-	const errorStatusCode = errorCodes[err.message] || 500;
-	const message = messages[err.message];
-	const gitHubProduct = getCloudOrServerFromGitHubAppId(res.locals.gitHubAppId);
-	const tags = { status: String(errorStatusCode), gitHubProduct };
-
-	statsd.increment(metricError.githubErrorRendered, tags, {});
-
-	return res.status(errorStatusCode).render("error.hbs", {
-		title: "GitHub + Jira integration",
-		errorReference,
-		message,
-		ctaUrl: req.query.retryUrl ? createUrlWithQueryString(req, req.query.retryUrl as string) : null,
-		nonce: res.locals.nonce,
-		githubRepoUrl: envVars.GITHUB_REPO_URL
+
+	// Sentry error middleware
+	router.use(Sentry.Handlers.errorHandler());
+
+	// Error and Catch all route - Handle anything that's missing or has throw an error
+	router.use((err: Error, req: Request, res: Response, next: NextFunction) => {
+		const errorReference = uuidv4();
+
+		req.log.error({ payload: req.body, errorReference, err, req, res }, "Error in frontend");
+
+		if (!isNodeProd() && res.locals.showError) {
+			return next(err);
+		}
+
+		// Check for IP Allowlist error from Github and set the message explicitly
+		// to be shown to the user in the error page
+		// TODO: this should probably be a util function
+		if (err.name == "HttpError" && err.message?.includes("organization has an IP allow list enabled")) {
+			err.message = Errors.IP_ALLOWLIST_MISCONFIGURED;
+		}
+
+		// TODO: move this somewhere else, enum?
+		const errorCodes = {
+			Unauthorized: 401,
+			Forbidden: 403,
+			"Not Found": 404
+		};
+		errorCodes[Errors.MISSING_JIRA_HOST] = 400;
+		errorCodes[Errors.MISSING_GITHUB_TOKEN] = 400;
+		errorCodes[Errors.MISSING_GITHUB_APP_NAME] = 400;
+		errorCodes[Errors.MISSING_ISSUE_KEY] = 400;
+
+		const messages = {
+			[Errors.MISSING_JIRA_HOST]: "Session information missing - please enable all cookies in your browser settings.",
+			[Errors.IP_ALLOWLIST_MISCONFIGURED]: `The GitHub org you are trying to connect is currently blocking our requests. To configure the GitHub IP Allow List correctly, <a href="${envVars.GITHUB_REPO_URL}/blob/main/docs/ip-allowlist.md">please follow these instructions</a>.`,
+			[Errors.MISSING_GITHUB_APP_NAME]: "There was a problem creating your GitHub App. Please make sure you filled the GitHub App name and try again."
+		};
+
+		const errorStatusCode = errorCodes[err.message] || 500;
+		const message = err instanceof UIDisplayableError ? err.message : messages[err.message];
```

Comment:
This is the only bit changed within this block, that display error suitable for frontend.